### DISCUSSION TITLE
Send BTC

### DISCUSSION THREAD

**Discussion by Jayke770** [1200wd/bitcoinlib#373] [Q&A] [answered]
I'm sending  BTC to another address how can I change/override the change address?

Tx Link: https://blockchair.com/bitcoin/testnet/transaction/964543edcaaeeecd8d77a41046369c2493b525779f1417007058b0b0f95d0228
![image](https://github.com/1200wd/bitcoinlib/assets/59675743/50380bc4-ac9a-4966-b168-bd0b507b2efb)

**Comment by mccwdev** [comment, marked as answer]
You can spend a single utxo and specify all output addresses like this:

``` python
wallet = wallet_delete_if_exists('example_btctestnet', force=True)
w = Wallet.create('example_btctestnet', network='bitcoinlib_test')
w.utxos_update()
t = w.send([('blt1qwx3vf0jlwpxvc5cl9g6gj4sltpluqem9jt0cde', 50000000),
            ('blt1qlkwskkwudqgee9az2cjqgncns5phykwx6eav3t', 49999000)], fee=1000, max_utxos=1)
t.info()
```

You can also use the input_arr or input_key_id argument to use specific utxo's

**Comment by mikeywaites** [comment]
Hey @mccwdev . hoping you can build on your answer to help me understand how this works.

I have this simple code example

```
def send_tx():
    # Create wallet
    recipient_address = "bc1pqmeagf32qlcyd8ehhuwzfyj5r3mmfqe75uuxw53cz3g7pxt8cafqphwjch"
    amount = 8000
    wallet = wallet_create_or_open(
        "dev", seed_phrase, network="bitcoin", witness_type="p2sh-segwit"
    )

    wallet.scan(account_id=0)

    t = wallet.send_to(
        recipient_address,
        amount,
        offline=True,
        account_id=0,
        number_of_change_outputs=1,
    )

    print(t.info())
```

This creates a transaction like i'd expect.  
```
Transaction 9d835c6a3456c0d12fea5b6efd187ddfdf35c701aed9b1012eccdbeaf942e89b
Date: None
Network: bitcoin
Version: 1
Witness type: segwit
Status: new
Verified: True
Inputs
- 37R8sDNcVYjyAr4mcPrDQoSiuXtwyUF3y9 0.02123440 BTC 300fdbf0753e4acc001f56e0ac17c2ed07b96378a4a06c3b3e6468726c2b501c 0
  p2sh-segwit sig_pubkey; sigs: 1 (1-of-1) valid
Outputs
- bc1pqmeagf32qlcyd8ehhuwzfyj5r3mmfqe75uuxw53cz3g7pxt8cafqphwjch 0.00008000 BTC p2tr U
- 36mZ2dqYZJNe5DH1m8Ze3kXkn4jVq1ScKq 0.02112400 BTC p2sh U
Size: 258
Vsize: 177
Fee: 3040
Confirmations: None
Block: None
Pushed to network: False
Wallet: dev
```

What i cannot for the life of me figure out is, how do i have the output to send the unspent UTXOs back to my wallet go back to the same address that the Inputs came from.

To be clear, this address is the address with the inputs, i want everything i dont spent to then go back to this address.
```
37R8sDNcVYjyAr4mcPrDQoSiuXtwyUF3y9
```

so that the transaction info looks like this


```
Transaction 9d835c6a3456c0d12fea5b6efd187ddfdf35c701aed9b1012eccdbeaf942e89b
Date: None
Network: bitcoin
Version: 1
Witness type: segwit
Status: new
Verified: True
Inputs
- 37R8sDNcVYjyAr4mcPrDQoSiuXtwyUF3y9 0.02123440 BTC 300fdbf0753e4acc001f56e0ac17c2ed07b96378a4a06c3b3e6468726c2b501c 0
  p2sh-segwit sig_pubkey; sigs: 1 (1-of-1) valid
Outputs
- bc1pqmeagf32qlcyd8ehhuwzfyj5r3mmfqe75uuxw53cz3g7pxt8cafqphwjch 0.00008000 BTC p2tr U
- 37R8sDNcVYjyAr4mcPrDQoSiuXtwyUF3y9 0.02112400 BTC p2sh U
Size: 258
Vsize: 177
Fee: 3040
Confirmations: None
Block: None
Pushed to network: False
Wallet: dev
```

appreciate any help

**Comment by mccwdev** [comment]
You have to send function and not the send_to, supply a list of addresses and the fee. The total value output of the addresses plus the fee has to be the same as the input value, then no extra change address will be created.

In the example above there are two outputs and a fee of 1000, which adds up to 100000000, which is equal to one test utxo.

**Reply by mikeywaites** [reply]
@mccwdev thanks for getting back to me.

is there an example of the above you could point me to or possibly provide some code example

**Comment by mccwdev** [comment]
Not sure what you are want to achieve, but this kind of address reuse is not good for privacy and safety.

However you could do something like

```python
w = wallet_create_or_open('wlt373', network='testnet')
w.scan(scan_gap_limit=1)
w.info()

output_arr = [
    ('tb1qyqjgxn8aycwgxjjgnzpen6u5z4tsmnyjhhf300', 400),
    ('tb1qplyhtmgdk0hjuevux22xvudvljkvw5cpgr4tkv', 400),
]

t = w.send(output_arr, fee=200, min_confirms=0, offline=False)
t.info()
```

Tx info: https://blocksmurfer.io/tbtc/transaction/1b533e9942974953974f3627c5151e767b14245765c9b12c56d1bbdaec09f6ed
